constructor/main.py

```diff
@@ -177,6 +180,31 @@ def main_build(
                     new_extras.append({orig: dest})
         info[extra_type] = new_extras
 
+    def has_frozen_file(extra_files: list[str | dict[str, str]]) -> bool:
+        def is_conda_meta_frozen(path_str: str) -> bool:
+            path = Path(path_str)
+            return path.parts == ("conda-meta", "frozen") or (
+                len(path.parts) == 4
+                and path.parts[0] == "envs"
```

Comment:
```suggestion
                len(path.parts) >= 4
                and path.parts[-3] == "envs"
```

I don't think the original works the way it's written for strings. In the loop above, all strings are converted into absolute paths, so `envs/default/conda-meta/frozen` will not be caught (only inside a mapping).

We can either change it to the suggestion here (but then miss `not/an/env/envs/<some name>/conda-meta/frozen`); or the code here must be moved before this for-loop: https://github.com/conda/constructor/blob/d992bd7fbda7958c2d1d1d279a13727d555d7c87/constructor/main.py#L165-L178

I suggest the latter. @jaimergp , thoughts?